src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -377,7 +377,12 @@ protected CompletableFuture<FetchedValue> fetchField(ExecutionContext executionC
         MergedField field = parameters.getField();
         GraphQLObjectType parentType = (GraphQLObjectType) parameters.getExecutionStepInfo().getUnwrappedNonNullType();
         GraphQLFieldDefinition fieldDef = getFieldDef(executionContext.getGraphQLSchema(), parentType, field.getSingleField());
-        GraphQLCodeRegistry codeRegistry = executionContext.getGraphQLSchema().getCodeRegistry();
+        return fetchField(fieldDef, executionContext, parameters);
+    }
+
+    private CompletableFuture<FetchedValue> fetchField(GraphQLFieldDefinition fieldDef, ExecutionContext executionContext, ExecutionStrategyParameters parameters) {
+        MergedField field = parameters.getField();
+        GraphQLObjectType parentType = (GraphQLObjectType) parameters.getExecutionStepInfo().getUnwrappedNonNullType();
 
```

Comment:
The old code is left in place

```
  protected CompletableFuture<FetchedValue> fetchField(ExecutionContext executionContext, ExecutionStrategyParameters parameters) {
        MergedField field = parameters.getField();
        GraphQLObjectType parentType = (GraphQLObjectType) parameters.getExecutionStepInfo().getUnwrappedNonNullType();
        GraphQLFieldDefinition fieldDef = getFieldDef(executionContext.getGraphQLSchema(), parentType, field.getSingleField());
        return fetchField(fieldDef, executionContext, parameters);
    }

    private CompletableFuture<FetchedValue> fetchField(GraphQLFieldDefinition fieldDef, ExecutionContext executionContext, ExecutionStrategyParameters parameters) {

```

and the engine only now calls the private method here.  This means we dont break backwards compatibility.  So nominally a 👍 

HOWEVER - we have long had the debate inside the team about JUST how much API contract we should have in the ExecutionStrategy land - eg can other compose their own engines based on using the base classes and I think the answer is... they really cant make whole new engines using this class via inheritance and hence perhaps these should not be API per say

So it might be prudent @andimarek to just change the method signature - since the old method is dangling and never called by the engine now

Reply:
Yeah, with the class marked `@PublicSpi` I didn't want to mess with the signatures, and three remaining references internally:

<img width="1044" alt="Screenshot 2024-02-28 at 5 23 59 pm" src="https://github.com/graphql-java/graphql-java/assets/1479220/a1511884-49c1-4cd1-a021-873d4f640ac5">